mlflow/anthropic/autolog.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+
+import mlflow
+import mlflow.anthropic
+from mlflow.entities import SpanType
+from mlflow.utils.autologging_utils.config import AutoLoggingConfig
+
+
+def _get_span_type(task_name: str) -> str:
+    span_type_mapping = {
+        "create": SpanType.CHAT_MODEL,
+    }
+    return span_type_mapping.get(task_name, SpanType.UNKNOWN)
+
+
+def construct_full_inputs(func, *args, **kwargs):
+    signature = inspect.signature(func)
+    # this does not create copy. So values should not be mutated directly
+    arguments = signature.bind_partial(*args, **kwargs).arguments
+
+    if "self" in arguments:
+        self = arguments.pop("self")
+
+        if hasattr(self, "model_name"):
+            arguments["model_name"] = self.model_name
```

Comment:
is this effectively adding a kwarg to the arguments? will there be any impact when it's passed to the original function?

Reply:
@daniellok-db The `argument` itself is a dictionary created from `bind_partial` that wraps args, kwargs, so it's fine to add/remove new keys. However, mutating the values (e.g. `arguments["messages"][0]["content"] = "abc"`) is should not be permitted.

Ref: https://github.com/mlflow/mlflow/pull/13801#discussion_r1845788420

Reply:
@gabrielfu That said, do we need to add `"model_name"` manually here? The kwargs should already contains `"model"`.

Reply:
@B-Step62 you're right, that's redundant